Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/JavaCropAlgorithm.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools.helper
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.graphics.Rect
+import android.util.Log
+
+private const val TAG = "cropAlgorithmSnail"
+
+class JavaCropAlgorithm : CropAlgorithm {
+    private lateinit var bounds: Rect
+    private lateinit var bitmap: Bitmap
+
+    private fun Rect.rowRange() = this.top..this.bottom
```

Comment:
Sorry, since we use x and y here, please refactor it respectively.
```suggestion
    private fun Rect.yRange() = this.top..this.bottom
```

Reply:
hmm.. ok

Reply:
done.